src/coordinator/ts/values.go

```diff
@@ -133,3 +141,51 @@ func NewFixedStepValues(millisPerStep time.Duration, numSteps int, initialValue
 		values:        values,
 	}
 }
+
+// At every resolution timestep, the last sample value is chosen
+func RawPointsToFixedStep(datapoints Datapoints, start time.Time, end time.Time, interval time.Duration) (FixedResolutionMutableValues, error) {
+	if end.Before(start) {
+		return nil, fmt.Errorf("start cannot be after end, start: %v, end: %v", start, end)
+	}
+
+	if interval == 0 {
+		return nil, errors.New("interval cannot be 0")
+	}
+
+	var numSteps int
+	if end == start {
+		numSteps = 1
+	} else {
+		numSteps = int(end.Sub(start) / interval)
+	}
+
+	fixStepValues := newFixedStepValues(interval, numSteps, math.NaN(), start)
+	fixedResIdx := 0
+	dpIdx := 0
+	numPoints := len(datapoints)
+	for t := start; !t.After(end) && fixedResIdx < numSteps; t = t.Add(interval) {
+		// Find first datapoint not before time t
+		for ; dpIdx < numPoints; dpIdx++ {
+			if !datapoints.DatapointAt(dpIdx).Timestamp.Before(t) {
+				break
+			}
+		}
+
+		if dpIdx >= numPoints {
+			fixStepValues.values[fixedResIdx] = math.NaN()
```

Comment:
nit: worth shortcutting?
```golang
{
 for ; currIdx < numSteps; currIdx++ {
  fixStepValues.values[currIdx] = math.NaN()
 }
 break
}
```

Reply:
oh yeah, actually that's not needed. I added a comment to make it clear